fiftyone/brain/internal/core/uniqueness.py

```diff
@@ -139,26 +151,30 @@ def compute_uniqueness(
     logger.info("Uniqueness computation complete")
 
 
-def _compute_uniqueness(embeddings, metric="euclidean"):
-    K = 3
-
-    num_embeddings = len(embeddings)
-    if num_embeddings <= K:
-        return [1] * num_embeddings
-    elif num_embeddings <= _MAX_PRECOMPUTE_DISTS:
-        embeddings = skm.pairwise_distances(embeddings, metric=metric)
-        metric = "precomputed"
-    else:
-        logger.info(
-            "Computing neighbors for %d embeddings; this may take awhile...",
-            num_embeddings,
+def _compute_uniqueness(
+    embeddings, similarity_index=None, metric="euclidean", n_neighbors=3
+):
+    if similarity_index:
+        _, dists_list = similarity_index._kneighbors(
+            query=embeddings, k=n_neighbors + 1, return_dists=True
```

Comment:
To clarify, you mean the retrieval is not parallelized for each embedding right? i.e. [this](https://github.com/voxel51/fiftyone-brain/blob/2374ef5b20985ad65b38665d55effd1cce725a7a/fiftyone/brain/internal/core/pinecone.py#L550) loop will be really slow. Even if this is the case I'd argue slow is better than segfaulting which is what the sklearn backend will do with any reasonably large dataset :)

Reply:
Leaving the optimization of similarity backends for later.

Reply:
Segfault after 15 seconds is arguably better than a loop that takes many hours to complete (for example). Either way the conclusion is that the feature doesn't support that size of dataset

Reply:
What are you proposing here? I am unclear on whether the expectation is that optimizations to using non-sklearn backends be made before this PR is merged or after.

Reply:
I'm just proposing that you test a non-sklearn backend on some larger datasets and see how long it takes for this sequential k-neighbors query implementation to run. If it takes a long time, we should consider one of three things:
1. Look into batching k-neighbors queries to make it more performant
2. Disable non-sklearn backends completely
3. Raise an error or at least log a "this may take awhile" warning above a certain dataset size

Reply:
I would be in favor of 1 or 3. I'd be happy to wait for the method to complete if I knew how long it might take. Batching knn queries would require changes to each of the different vector backends so I think those changes should be a separate PR.

Reply:
Regarding `see how long it takes for this sequential k-neighbors query implementation to run` -- I am working on benchmarking this on `places` dataset.

_Look into batching k-neighbors queries to make it more performant_ would be the best option imo and requires the most amount of changes and testing. I can add a warning for slow computation time until we are done working on 1.